File: php/features.php
<?php
declare(strict_types=1);

// N404ErrorHandler SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class N404ErrorHandlerFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new N404ErrorHandlerBaseFeature();
            case "test":
                return new N404ErrorHandlerTestFeature();
            default:
                return new N404ErrorHandlerBaseFeature();
        }
    }
}

File: py/features.py
# N404ErrorHandler SDK feature factory

from feature.base_feature import N404ErrorHandlerBaseFeature
from feature.test_feature import N404ErrorHandlerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: N404ErrorHandlerBaseFeature(),
        "test": lambda: N404ErrorHandlerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
